try/run_central_instances.py

```diff
@@ -0,0 +1,2341 @@
+                    format(self._filename, type(data)))
+
+            servers = data.get('servers', None)
+            if servers is None:
+                raise ServerDefinitionFileError(
+                    "The WBEM server definition file {0!r} does not define a "
+                    "'servers' item, but items: {1}".
+                    format(self._filename, data.keys()))
+            if not isinstance(servers, OrderedDict):
+                raise ServerDefinitionFileError(
+                    "The 'servers' in WBEM server definition file {0!r} "
+                    "must be a dictionary, but is {1}".
+                    format(self._filename, type(servers)))
+            self._servers.update(servers)
+
+            server_groups = data.get('server_groups', OrderedDict())
+            if not isinstance(server_groups, OrderedDict):
+                raise ServerDefinitionFileError(
+                    "The 'server_groups' in WBEM server definition file "
+                    "{0!r} must be a dictionary, but is a {1}".
+                    format(self._filename, type(server_groups)))
+            for sg_nick in server_groups:
+                visited_sg_nicks = list()
+                self._check_sg(sg_nick, server_groups, servers,
+                               visited_sg_nicks)
+            self._server_groups.update(server_groups)
+
+    def _check_sg(self, sg_nick, server_groups, servers, visited_sg_nicks):
+        visited_sg_nicks.append(sg_nick)
+        server_group = server_groups[sg_nick]
+        if not isinstance(server_group, list):
+            raise ServerDefinitionFileError(
+                "Server group {0!r} in WBEM server definition file {1!r} "
+                "must be a list, but is a {2}".
+                format(sg_nick, self._filename, type(server_group)))
+        for nick in server_group:
+            if nick in visited_sg_nicks:
+                raise ServerDefinitionFileError(
+                    "Item {0!r} in server group {1!r} in WBEM server "
+                    "definition file {2!r} references a server group in its "
+                    "ancestry".
+                    format(nick, sg_nick, self._filename))
+            if not isinstance(nick, str):
+                raise ServerDefinitionFileError(
+                    "Item {0!r} in server group {1!r} in WBEM server "
+                    "definition file {2!r} must be a string, but is a {3}".
+                    format(nick, sg_nick, self._filename, type(nick)))
+            if nick in server_groups:
+                self._check_sg(nick, server_groups, servers, visited_sg_nicks)
+            elif nick not in servers:
+                raise ServerDefinitionFileError(
+                    "Item {0!r} in server group {1!r} in WBEM server "
+                    "definition file {2!r} is not a known server or server "
+                    "group".
+                    format(nick, sg_nick, self._filename))
+
+    @property
+    def filename(self):
```

Comment:
I think the `filename()` method should be added to the version of the class in server_file.py.